prboom2/src/r_defs.h

```diff
@@ -283,13 +289,20 @@ typedef enum
   ST_NEGATIVE
 } slopetype_t;
 
-typedef byte r_flags_t;
-#define RF_TOP_TILE 0x01 // Upper texture needs tiling
-#define RF_MID_TILE 0x02 // Mid texture needs tiling
-#define RF_BOT_TILE 0x04 // Lower texture needs tiling
-#define RF_IGNORE   0x08 // Renderer can skip this line
-#define RF_CLOSED   0x10 // Line blocks view
-#define RF_ISOLATED 0x20 // Isolated line
+typedef uint16_t r_flags_t;
```

Comment:
Please use a regular unsigned short here. Specifying 16 bits prevents the compiler from choosing on its own, which can only reduce performance. Of course, that's unlikely to make any difference here, but there's no need to add this restriction. Please also remove the use of the colon operator elsewhere for the same reason - unless you can demonstrate that choosing for the compiler creates a real impact, I don't think it's worth adding this pattern (which nothing else in the code base uses as far as I know). There _are_ places that do specify exact sizes, but usually this is a safety feature to deal with the binary formats (though there may be other uses to clean up at some point).

Reply:
> Specifying 16 bits prevents the compiler from choosing on its own, which can only reduce performance

There are tens or hundreds of thousands of these structures in memory, so choosing the smallest type that fits the data is important, both for memory usage and cache efficiency.

> Please also remove the use of the colon operator elsewhere for the same reason

I also used bitfields to keep the size of `vertex_t` as low as possible.  Again, there can be hundreds of thousands of these in memory at once, and cache performance is going to be important.

Reply:
Supported maps are unimaginably small in the scope of modern gaming and those kinds of optimizations are completely irrelevant. If anything I've seen these attempts get in the way of optimization by blocking the compiler. Someone tried to optimize zone memory for the cache and low and behold gutting all those attempts actually made the zone memory faster. If you have definitive evidence let me know. Udmf alone added tons of integers to these data structures.

Reply:
I can remove bitfields.  I'll also keep `uint16_t` since gaining nanoseconds of speed on some hypothetical system where `unsigned short` is not `uint16_t` and somehow faster is an irrelevant optimization.